request.go

```diff
@@ -291,6 +292,37 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 					models = reflect.Append(models, m)
 				}
 
+				if links != nil {
+					var assignedValue bool
+					linkModel := reflect.New(fieldValue.Type().Elem().Elem())
+					linkModelValue := linkModel.Elem()
+					linkModelType := linkModelValue.Type()
+
+					for i := 0; i < linkModelValue.NumField(); i++ {
```

Comment:
We are iterating through the model because this is the only way to find the attribute on the struct that has the `jsonapi:"links"` tag. So once we find it, we unmarshal it onto the model (the Resource struct).